dunwoody_disaster/CharacterFactory.py

```diff
@@ -23,6 +29,30 @@ def __init__(self):
         self.weapons = {}
         self.defenses = {}
 
+    def set_health(self, health: int):
+        self.curHealth = health
+        if self.maxHealth == 0:
+            percentage = 0
+        else:
+            percentage = (health // self.maxHealth) * 100
+        self.health_meter.setPercentage(percentage)
```

Comment:
This should also update the health value in the `CharacterState`, The same for magic and mechanical.